### DISCUSSION TITLE
adding node and node attributes iteratively.

### DISCUSSION THREAD

**Discussion by Jeriousman** [networkx/networkx#5362] [Q&A]
I want to add node not one by one but at once. I have pandas dataframe data here.
```
G.add_nodes_from((final_carddata['cardKey']) 
```
I know this will add nodes at once all. But if I do as below, nodes are well added iteratively but node features are all just copied and added the whole value in every node. But I want to make it iteratively just like the nodes.
``` 
G.add_nodes_from(final_carddata['cardKey'] , node_type = 'card', node_feature = final_carddata['node_feature']
```

Can anyone provide how to do what I want in code?

Thank you in advance.

**Comment by lior8** [comment]
Try taking a look [here](https://networkx.org/documentation/stable/reference/classes/generated/networkx.Graph.add_nodes_from.html) and see if that solves your problem. If I understood you correctly, it should be what you need.

The last example in there suggests a way to add all nodes at once, each with their own respective data.
